src/voloctree/voloctree_mapper.cpp

```diff
@@ -1237,13 +1237,21 @@ bool VolOctreeMapper::_recoverPartition()
     for (int reference_rank : toreference_rank[m_rank]) {
         uint64_t reference_first_morton = partitionFDReference[reference_rank];
         uint64_t reference_last_morton = partitionLDReference[reference_rank];
+        // Initialize idx position in mapped tree to the first mapped
+        // octant inside the overlapping region
         while (morton < reference_first_morton) {
             idx++;
             if (idx == mappedPatch->getTree().getNumOctants()) {
                 break;
             }
             morton = mappedPatch->getTree().getLastDescMorton(idx);
         }
```

Comment:
Would it make sense to set the morton to an invalid number when idx reaches the end of the octant list?
```
        ...
        while (morton < reference_first_morton) {
            idx++;
            if (idx == mappedPatch->getTree().getNumOctants()) {
                morton = PABLO::INVALID_MORTON;
                break;
            }
            morton = mappedPatch->getTree().getLastDescMorton(idx);
        }
        ...
```
(The new check can then test the Morton number being valid.)

Hopefully this is not needed, but it may may help debugging possible problems in the same function.

Reply:
I think it's clearer in this way. The loop is reaching the final octant of the structure, not an invalid element. The second while is to skip if the index is at the final stage of the octant container, putting an invalid morton could be misunderstanding. 
If the check on num. octants is not clear (in my opinion it is) we could put a boolean variable, identifying the reaching of the end of the octants container and to be checked it before entering in the second while loop, of an old-style, but always used for nested loop, _goto_ statement at the place of the first break.